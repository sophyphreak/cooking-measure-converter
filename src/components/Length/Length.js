import React from 'react';
import TableRow from '../TableRow/TableRow';

const Length = ({ lengthProps }) => {
  const { length, onImperialLengthChange, onMetricLengthChange } = lengthProps;
  const { imperialLength, imperialUnit, metricLength, metricUnit } = length;
  return (
    <TableRow
      metricUnit={metricUnit}
      metricValue={metricLength}
      onMetricChange={onMetricLengthChange}
      imperialValue={imperialLength}
      onImperialChange={onImperialLengthChange}
      imperialUnit={imperialUnit}
    />
  );
};

export default Length;
