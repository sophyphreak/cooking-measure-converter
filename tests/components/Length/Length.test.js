import React from 'react';
import { shallow } from 'enzyme';
import Length from '../../../src/components/Length/Length';

let length, onImperialLengthChange, onMetricLengthChange, lengthProps, wrapper;

beforeEach(() => {
  length = {
    imperialLength: '',
    imperialUnit: 'in',
    metricLength: '',
    metricUnit: 'cm'
  };
  onImperialLengthChange = jest.fn();
  onMetricLengthChange = jest.fn();
  lengthProps = {
    length,
    onImperialLengthChange,
    onMetricLengthChange
  };
  wrapper = shallow(<Length lengthProps={lengthProps} />);
});

test('should render Length correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
