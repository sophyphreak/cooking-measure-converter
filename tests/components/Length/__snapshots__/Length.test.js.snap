// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Length correctly 1`] = `
<TableRow
  imperialUnit="in"
  imperialValue=""
  metricUnit="cm"
  metricValue=""
  onImperialChange={[MockFunction]}
  onMetricChange={[MockFunction]}
/>
`;
